falcon/middleware.py

```diff
@@ -146,6 +157,13 @@ def process_response(
                 resp.set_header('Access-Control-Allow-Headers', allow_headers)
                 resp.set_header('Access-Control-Max-Age', '86400')  # 24 hours
 
+            has_request_private_network = (
```

Comment:
PERF: Let's switch these conditions around for performance.

First of all, the absolute majority of users will not use this new parameter, so first check against it. On the recent CPython versions, such a check is very fast.

I'm thinking roughly along the lines below
```python
    if self.allow_private_network:
        if req.get_header('Access-Control-Request-Private-Network'):
            resp.set_header('Access-Control-Allow-Private-Network', 'true')
```
(These conditions can also be joined by `and`, but please keep `self.allow_private_network` first.)